neurodiffeq/solvers.py

```diff
@@ -347,8 +378,25 @@ def closure(zero_grad=True):
 
                 # accumulate gradients before the current graph is collected as garbage
                 if key == 'train':
-                    loss.backward()
+                    loss.backward(retain_graph=True, inputs = list(chain.from_iterable(n.parameters() for n in self.nets)))
```

Comment:
Why specifying `inputs`? Are we excluding something (e.g. parameters from the `NNGenerator`)
Why set `retain_graph` to True?